### PR TITLE
[Regression] Add linear regression

### PR DESCRIPTION
**Pull request by RedGl0w**
This fixes https://github.com/numworks/epsilon/issues/1529
Todo :
- [ ] Add translations
- [x] Add the display of r and r^2 in the calculation controller for affine regression

### CODE REVIEW

**Comment by adriweb** [comment]
Hm. Linear is supposed to be `a*x` and affine `a*x+b`.
Looks like it's reversed here, and the real affine is just already implemented under the name linear.

**Comment by RedGl0w** [comment]
> Hm. Linear is supposed to be `a*x` and affine `a*x+b`.
> Looks like it's reversed here, and the real affine is just already implemented under the name linear.

Oh fu** I haven't realized when I did that :sob:
I hope my math teacher will never see that :troll:
I'll patch that in a few minutes !
Thank you !

File: apps/regression/base.universal.i18n [hunk 1]
```diff
@@ -1,3 +1,4 @@
+AffineRegressionFormula  = " y=a·x "
```
Comment (LeaNumworks) [inline apps/regression/base.universal.i18n:1]: This is the LinearRegressionFormula!

`LinearRegressionFormula  = " y=a·x+b "` is defined in `apps/shared.universal.i18n` but should be renamed AffineRegressionFormula there and in `apps/graph/graph/banner_view.cpp`, and then used in the affine model.

File: apps/regression/model/affine_model.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#include "linear_model.h"
```
Comment (LeaNumworks) [inline apps/regression/model/affine_model.cpp:1]: Should include affine_model.h

File: apps/regression/model/affine_model.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#ifndef REGRESSION_LINEAR_MODEL_H
```
Comment (LeaNumworks) [inline apps/regression/model/affine_model.h:1]: The header guardian should be REGRESSION_AFFINE_MODEL_H

File: apps/regression/base.universal.i18n (hunk 1 above)
Comment (RedGl0w) [reply apps/regression/base.universal.i18n:1]: Yes the switch between affine and linear was a bit chaotic

File: apps/regression/model/linear_model.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#include "linear_model.h"
+#include "affine_model.h"
```
Comment (LeaNumworks) [inline apps/regression/model/linear_model.cpp:1]: linear_model.cpp should include linear_model.h

File: apps/regression/model/linear_model.cpp [hunk 5]
```diff
@@ -28,7 +27,7 @@ double LinearModel::levelSet(double * modelCoefficients, double xMin, double ste
   if (a == 0) {
     return NAN;
   }
-  return (y-b)/a;
+  return y-b;
```
Comment (LeaNumworks) [inline apps/regression/model/linear_model.cpp:30]: There is no b coefficient in the linear model, so the line `double b = modelCoefficients[1];` should be removed.
The levelSet is supposed to give x given y, so here it should return y/a

File: apps/regression/model/linear_model.cpp [hunk 6]
```diff
@@ -28,7 +27,7 @@ double LinearModel::levelSet(double * modelCoefficients, double xMin, double ste
   if (a == 0) {
```
Comment (LeaNumworks) [inline apps/regression/model/linear_model.cpp:27]: As you are modifying this file anyways, could you please change this to `a == 0.0` ? Thanks!

File: apps/regression/model/linear_model.cpp (hunk 5 above)
Comment (LeaNumworks) [reply apps/regression/model/linear_model.cpp:30]: A few lines below, in fit, please remove `modelCoefficients[1] = store->yIntercept(series);`

Comment (LeaNumworks) [reply apps/regression/model/linear_model.cpp:30]: And again in `LinearModel::partialDerivate`, the case
` if (derivateCoefficientIndex == 1) {
    // Derivate: 1;
    return 1;
  }`
should not exist

File: apps/regression/model/linear_model.h [hunk 7]
```diff
@@ -1,5 +1,5 @@
-#ifndef REGRESSION_LINEAR_MODEL_H
-#define REGRESSION_LINEAR_MODEL_H
+#ifndef REGRESSION_AFFINE_MODEL_H
```
Comment (LeaNumworks) [inline apps/regression/model/linear_model.h:1]: The header guardians should match the file name, so REGRESSION_LINEAR_MODEL_H here

**Comment by LeaNumworks** [comment]
Hi, thanks for the changes!

**Comment by LeaNumworks** [comment]
I took the liberty to rebase your PR, so I push-forced it.

We decided to keep the linear regression as aX+b and to rename aX as Proportional regression. 
I also added commits that do small cleanings and add a test.
I removed r and r2, we will do it later.

Thanks again!

**Pull request merged**